src/modules/product/controller/productController.ts

```diff
@@ -46,4 +46,21 @@ const createShop = async (req: ExtendRequest, res: Response) => {
       });
     }
   };
-export default { createProduct, createShop }
\ No newline at end of file
+
+  const updateProductAvailability = async (req: ExtendRequest, res: Response) => {
+    try {
+        await productRepositories.markExpiredProducts();

```

Comment:
This logic seems working in the wrong way. Please review it and fix. Otherwise you will be updating all products in the **`Products Table`**.